state-chain/chains/src/btc.rs

```diff
@@ -958,22 +967,32 @@ impl<T: SerializeBtc> SerializeBtc for &[T] {
 #[derive(Encode, Decode, TypeInfo, MaxEncodedLen, Clone, RuntimeDebug, PartialEq, Eq)]
 pub enum BitcoinOp {
 	PushUint { value: u32 },
-	PushBytes { bytes: BoundedVec<u8, ConstU32<MAX_SEGWIT_PROGRAM_BYTES>> },
+	PushBytes { bytes: BoundedVec<u8, ConstU32<MAX_PUSHABLE_BYTES>> },
 	Drop,
 	CheckSig,
 	Dup,
 	Hash160,
 	EqualVerify,
 	Equal,
-	// Not part of the bitcoin spec, implemented for convenience
+	Return,
+	// The following are not part of the bitcoin spec, implemented for convenience:
 	PushArray20 { bytes: [u8; 20] },
 	PushArray32 { bytes: [u8; 32] },
 	PushVersion { version: u8 },
 }
 
-#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, RuntimeDebug, PartialEq, Eq)]
+#[derive(Encode, Decode, TypeInfo, Clone, RuntimeDebug, PartialEq, Eq)]
 pub struct BitcoinScript {
-	bytes: BoundedVec<u8, ConstU32<MAX_BITCOIN_SCRIPT_LENGTH>>,
+	bytes: Vec<u8>,
+}
+
+// We only store (spendable) BitcoinScripts that we authored ourselves.
+// For these, the max encoded length is MAX_BITCOIN_SCRIPT_LENGTH.
+// For anything else, we don't need MaxEncodedLen.
```

Comment:
I noticed that PushBytes allows MAX_PUSHABLE_BYTES, but we were storing it in a bounded vec of size MAX_BITCOIN_SCRIPT_LENGTH. 

The only reason we defined this was to derive MaxEncodedLen, so I moved the definition of this here.